msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -1179,14 +1164,9 @@ class NativeAuthPublicClientApplicationKotlinTest : PublicClientApplicationAbstr
         val result = application.signIn(username, password)
         assertTrue(result is SignInResult.Complete)
 
-        try {
-            application.resetPassword(username)
-        } catch (exception: MsalException) {
-            assertEquals(MsalClientException.INVALID_PARAMETER, exception.errorCode)
-            assertEquals("An account is already signed in.", exception.message)
-            return@runBlocking
-        }
-        fail() // An exception should happen
+        val newResult = application.resetPassword(username)
+        assertTrue(newResult is ResetPasswordError)
+        assertTrue((newResult as ResetPasswordError).exception?.message.equals("An account is already signed in."))
```

Comment:
Do you know of any scenarios that throw an exception deeper in the SDK? If so, could you write a test for that as well? To ensure `CommandDispatcher.submitSilentReturningFuture` and `rawCommandResult.checkAndWrapCommandResultType` logic work as expected as well. 

If there are no known scenarios where this happens, then feel free to resolve this.

Reply:
- BaseNativeAuthController throws ClientException. They could only be called in NativeAuthMsalController and we didn't.
- NativeAuthMsalController throws IOException::class, ClientException::class, ServiceException::class ArgumentException::class, ServiceException::class. mainly from acquireTokenSilent & renewAT used by AcquireTokenNoFixedScopesCommand (accountState.getAccessToken). It will be handled by is Exception in getAccessToken`
- AcquireTokenNoFixedScopesCommandParameters throws ArgumentException -> this parameter will be delted
- Request throws ClientException -> NativeAuthRequestHandlerTest include
- verifyNoUserIsSignedIn() throws MsalClientException

exception during command execution will be handled by CommandResultUtil.kt. The Exception has been wrapped as INativeAuthCommandResult.UnknownError and then return as generic error in the interface layer.
Except than getAccessToken, which has its own exception wrapper as below. The test for this part is missing.
```kotlin
return@withContext when (commandResult) {
                    is ServiceException -> {
                        GetAccessTokenError(
                            exception = ExceptionAdapter.convertToNativeAuthException(commandResult),
                            correlationId = "UNSET"
                        )
                    }

                    is Exception -> {
                        GetAccessTokenError(
                            exception = commandResult,
                            correlationId = "UNSET"
                        )
                    }
```

Reply:
- Request throws ClientException if empty arguments -> Add **testEmptyRequestParametersToGenericErrorNotThrownException()** 

handled by is Exception in getAccessToken` -> Not so easy (maybe also not suitable) to trigger SilentTokenCommandParameters.validate() to throw ArgumentException in integration test.